### DISCUSSION TITLE
Way to prevent upload to destination if more as one "actions-gh-pages" used?

### DISCUSSION THREAD

**Discussion by AlwinEsch** [peaceiris/actions-gh-pages#684] [Q&A] [answered]
With this https://github.com/AlwinEsch/test-xbmc/blob/master/.github/workflows/documentation-creation.yml#L37-L51 (temporary Repo for tests) I have the dilemma that **"pages build and deployment"** is started twice. It then breaks off the first again and brings it up correctly in the second, but I don't think it's so nice that unnecessary process work is then carried out (see also https://github.com/AlwinEsch/test-xbmc/actions).

Is there a way to avoid the upload in the first and only to do it in the second, or other ways to create two different documentation in one branch?

The background in Kodi is to bring in documentation for itself and suitable for add-ons, associated request here https://github.com/xbmc/xbmc/pull/20511.

If something exists and I overlooked it in the search, is there already an issue, discussion or something else?

**Comment by peaceiris** [comment, marked as answer]
We can run workflows sequentially.

```yaml
jobs:
  deploy:
    runs-on: ubuntu-20.04
    concurrency:
      group: ${{ github.workflow }}-${{ github.ref }}
    steps:
```

For more details:

- See the examples of the README.md on this project: https://github.com/peaceiris/actions-gh-pages#examples
- See the official docs of GitHub Actions: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#concurrency

**Comment by AlwinEsch** [comment]
I had forgotten correct `concurrency` and is now included, thanks for that!

Was just about a slightly different problem. I have now introduced it in a different way, so that `peaceiris/actions-gh-pages@v3` is only used once and using `sed` on the two Doxygen projects it is copied to the same folder for upload.
What was also possible was to do two independent workflow jobs in the yml, only then would be a lot of unnecessary work.

Maybe it would be a feature `publish_dir` to be able to use several folders, e.g. as 
```yml
publish_dir: ["./folder/xyz", "./another_folder/zyx"]
```
and all goes to `destination_dir`.
